README.md

```diff
@@ -1,8 +1,34 @@
 # Task
 
-Welcome to your new gem! In this directory, you'll find the files you need to be able to package up your Ruby library into a gem. Put your Ruby code in the file `lib/task`. To experiment with that code, run `bin/console` for an interactive prompt.
+Task provides a toolbox for generating, tracking and serializing tasks to be performed.
+It is NOT a job queuing framework, a la resque or qless, and does not provide ways to execute tasks
+in a distributed fashion. Rather, Task could be used with such a framework to provide "at-least-once"
+task execution guarantees, or to perform multiple, lightweight tasks within a single job.
 
-TODO: Delete this and the text above, and describe your gem
+An example task lifecycle could be follows:
+
+1. A task is generated by a task generator process
+2. The task is saved to a backing data store, recording that it has been generated and should be completed.
+3. The task is passed, in serialized format, directly from the generating process to another worker process
```

Comment:
I think it goes without saying that it has to be serialized somehow, and how is an implementation detail.